Basic/DistributedAuthoritySocialHub/Assets/Scripts/Input/Mobile/MobileGamepadBehaviour.cs

```diff
@@ -0,0 +1,178 @@
+        string m_LookAction;
+        [InputControl(layout = "Button"), SerializeField]
+        string m_JumpAction;
+        [InputControl(layout = "Button"), SerializeField]
+        string m_InteractAction;
+        [InputControl(layout = "Button"), SerializeField]
+        string m_SprintAction;
+        //[InputControl(layout = "Button"), SerializeField]
+        //string m_ToggleNetworkStatsAction;
+        [InputControl(layout = "Button"), SerializeField]
+        string m_MenuAction;
+
+        InputDevice m_Device;
+        Dictionary<string, InputControl> m_ControlMap = new();
+        MobileGamepadState m_RuntimeState;
+
+        async void OnEnable()
+        {
+            var isMobile = await InputSystemManager.IsMobile;
+            if (!isMobile)
+                return;
+
+            if (!TryGetDevice())
+                return;
+
+            SetupControlBindings();
+            m_RuntimeState = MobileGamepadState.GetOrCreate;
+            m_RuntimeState.ButtonStateChanged += SendControlEvent;
+            m_RuntimeState.JoystickStateChanged += SendControlUpdate;
+        }
+
+        void OnDisable()
+        {
+            if (m_Device != null)
+            {
+                if (m_Device.usages.Count == 1 && m_Device.usages[0] == "OnScreen")
+                    InputSystem.RemoveDevice(m_Device);
+
+                if (m_RuntimeState != null)
+                {
+                    m_RuntimeState.ButtonStateChanged -= SendControlEvent;
+                    m_RuntimeState.JoystickStateChanged -= SendControlUpdate;
+                }
+            }
+        }
+
+        bool TryGetDevice()
+        {
+            try
+            {
+                m_Device = InputSystem.GetDevice<Gamepad>();
+                if (m_Device == null)
+                {
+                    m_Device = InputSystem.AddDevice<Gamepad>();
+                    InputSystem.AddDeviceUsage(m_Device, "OnScreen");
+                }
+            }
+            catch (Exception exception)
+            {
+                Debug.LogError(
+                    $"Could not create device with layout 'Gamepad' used in '{GetType().Name}' component");
+                Debug.LogException(exception);
+                return false;
+            }
+
+            return true;
+        }
+
+        void SetupControlBindings()
+        {
+            m_ControlMap = new Dictionary<string, InputControl>
+            {
+                { nameof(MobileGamepadState.LeftJoystick), MapRuntimeControl(m_MoveAction) },
+                { nameof(MobileGamepadState.RightJoystick), MapRuntimeControl(m_LookAction) },
+                { nameof(MobileGamepadState.ButtonJump), MapRuntimeControl(m_JumpAction) },
+                { nameof(MobileGamepadState.ButtonInteract), MapRuntimeControl(m_InteractAction) },
+                { nameof(MobileGamepadState.ButtonSprint), MapRuntimeControl(m_SprintAction) },
+                // will re-visit to evaluate if this button is necessary
```

Comment:
do you want to revisit it in a later PR?

Reply:
Yeah, we'll revisit if we need this button in the UI icons PR.